### PR TITLE
Fix windows path unresolving

### LINKED ISSUE #11599: Groups in the mission selection got mixed up.
I guess it's the map category PR, but noidea if it really is. TD and RA bleed, might be even D2k.

### PR DESCRIPTION
**Pull request by pchote**
This should in principle fix #11599, but I do not have a suitable system to test with.
Ping @GraionDilach.

### CODE REVIEW

**Comment by pchote** [comment]
IMO this is probably too risky to take directly into a hotfix release without a proper playtest.  If others agree then it can move to Next + 1.

**Comment by GraionDilach** [comment]
Will test it during the weekend then.

**Comment by reaperrr** [comment]
This actually breaks campaign sorting for good on Windows. No idea why, though.

**Comment by RoosterDragon** [comment]
This PR plus the following works for me:

```c#
var previews = modData.MapCache
	.Where(p => p.Status == MapStatus.Available)
	.Select(p => new
	{
		Preview = p,
		Index = missionMapPaths.IndexOf(Platform.UnresolvePath(p.Package.Name).Replace('\\', '/'))
	})
	.Where(x => x.Index != -1)
	.OrderBy(x => x.Index)
	.Select(x => x.Preview);
```

On Windows, the unresolved path uses `\` as the path separator, this needs to be changed to `/` to match the config.

The ordering function also needs the same change - or you can rejig as above to avoid the duplication.

**Comment by pchote** [comment]
> On Windows, the unresolved path uses \ as the path separator, this needs to be changed to / to match the config.

We should move this into `UnresolvePath`

**Comment by pchote** [comment]
Updated.  Can you please test again @RoosterDragon?

File: OpenRA.Mods.Common/Widgets/Logic/MissionBrowserLogic.cs [hunk 1]
```diff
@@ -105,10 +105,10 @@ public MissionBrowserLogic(Widget widget, ModData modData, World world, Action o
 
 				foreach (var kv in yaml)
 				{
-					var missionMapPaths = kv.Value.Nodes.Select(n => Path.GetFullPath(n.Key)).ToList();
+					var missionMapPaths = kv.Value.Nodes.Select(n => n.Key).ToList();
 
 					var previews = modData.MapCache
-						.Where(p => p.Status == MapStatus.Available && missionMapPaths.Contains(p.Package.Name))
+						.Where(p => p.Status == MapStatus.Available && missionMapPaths.Contains(Platform.UnresolvePath(p.Package.Name)))
 						.OrderBy(p => missionMapPaths.IndexOf(p.Package.Name));
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Widgets/Logic/MissionBrowserLogic.cs]: Need to UnresolvePath here too or it won't order correctly.

**Comment by RoosterDragon** [comment]
Works - however, do we also need to do the opposite for `ResolvePath`? i.e. Change '/' into the platform specific directory char?

**Comment by pchote** [comment]
Shouldn't do - Windows happily consumes `/` for paths.

**Comment by pchote** [comment]
Updated.

**Review by RoosterDragon — DISMISSED** [review]

**Comment by reaperrr** [comment]
Needs rebase.

**Comment by pchote** [comment]
Rebased.

**Review by RoosterDragon — APPROVED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/a6d4a0c4ae783945f64d2261cfb809a74fc9ed42...213bfc949d982a7324a0fb6b55a026ca54860423)

**Pull request merged**